context.go

```diff
@@ -179,6 +179,10 @@ func (c *Ctx) resolveProjectRoot(path string) (string, error) {
 //
 // The second returned string indicates which GOPATH value was used.
 func (c *Ctx) SplitAbsoluteProjectRoot(path string) (string, error) {
+	if path == filepath.Join(c.GOPATH, "src") {
```

Comment:
I made the following changes:
Instead of comparing the `path` to `$GOPATH/src`, we start by ensuring we are actually within `$GOPATH/src` (through `internal.HasFilepathPrefix`).
If that's the case, we check if `path` is shorter than or equal to `srcprefix` (at most it could be shorter than `srcprefix` by the `len` of `filepath.Separator`, otherwise if it has a trailing slash it would be equal). If this is true, then the project root is `$GOPATH/src` and we return an error.

This should handle case sensitivity and trailing separators if my assumptions are correct. @sdboyer What do you think?